rp_gpconnect/tasks.py

```diff
@@ -0,0 +1,64 @@
+import os
+
+from celery.task import Task
+from celery.utils.log import get_task_logger
+from django.conf import settings
+from openpyxl import load_workbook
+from temba_client.v2 import TembaClient
+
+from sidekick.utils import get_whatsapp_contact_id
+
+from .models import ContactImport
+
+log = get_task_logger(__name__)
+
+
+class ProcessContactImport(Task):
```

Comment:
I would suggest making this a function based task, you can then easily implement a retry strategy.

Example:
https://github.com/praekeltfoundation/rp-sidekick/blob/develop/rp_asos/tasks.py#L252-L263

Maybe at a later stage you could separate this into two tasks, each with their own retry strategy. One main task for reading the excel that will start a sub task for each msisdn. This means that when a WA contact check fails on the last msisdn in the list, only that one will be retried.

Reply:
great idea! thanks